commons/commons-transpiler/src/test/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLineTest.kt

```diff
@@ -0,0 +1,66 @@
+package ac.uk.ebi.transpiler.cli
+
+import ac.uk.ebi.biostd.SubFormat
+import ac.uk.ebi.transpiler.service.FilesTableTemplateTranspiler
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import io.github.glytching.junit.extension.folder.TemporaryFolderExtension
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.verify
+import org.apache.commons.cli.HelpFormatter
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(MockKExtension::class, TemporaryFolderExtension::class)
+class TranspilerCommandLineTest(
+    private val temporaryFolder: TemporaryFolder,
+    @MockK private val mockHelpFormatter: HelpFormatter,
+    @MockK private val mockTranspiler: FilesTableTemplateTranspiler
+) {
+    private val testInstance = TranspilerCommandLine(mockHelpFormatter, mockTranspiler)
+
+    @BeforeEach
+    fun setUp() {
+        temporaryFolder.createFile("template.tsv")
+        every { mockHelpFormatter.printHelp(any(), any()) }.answers { nothing }
+        every { mockTranspiler.transpile("", listOf("colA", "colB"), "/path", "base", SubFormat.TSV) }.returns("pageTab")
+    }
+
+    @AfterEach
+    fun tearDown() = clearAllMocks()
+
+    @Test
+    fun `transpile template`() {
+        val args = arrayOf(
+            "-b", "base",
+            "-f", "TSV",
+            "-d", "/path",
+            "-c", "colA, colB",
+            "-t", "${temporaryFolder.root.absolutePath}/template.tsv")
+        val pageTab = testInstance.transpile(args)
+
+        assertThat(pageTab).isEqualTo("pageTab")
+        verify(exactly = 0) { mockHelpFormatter.printHelp(any(), any()) }
+    }
+
+    @Test
+    fun `missing arguments`() {
+        val pageTab = testInstance.transpile(arrayOf("-f", "JSON"))
+
+        assertThat(pageTab).isEqualTo("")
+        verify(exactly = 1) { mockHelpFormatter.printHelp(any(), any()) }
```

Comment:
personally unless not possible I prefer to assert mocks with specific values as `any` may hide a bug in the code

Reply:
Done